pkg/cli/codearea.go

```diff
@@ -207,9 +261,18 @@ func (w *codeArea) handleKeyEvent(key ui.Key) bool {
 		}
 		return true
 	}
+
+	// Perform abbreviation expansion if a space, newline, pipe or punctuation
```

Comment:
A reader would expect the code to expand abbreviations and word abbreviations to be at the same place, but they aren't.

It's worth spending some time refactoring the code so that it is the case.

Reply:
It's not clear that doing so would be an improvement. The reason for the current structure is due to treating a newline char as whitespace that triggers a word expansion. The case statements in the `switch key` block below won't be executed if, as currently happens, the newline is handled by the `if w.OverlayHandler.Handle(term.KeyEvent(key))` block.

I'm going to rebase master and address your other points. Then we can revisit this if you still feel strongly that unifying the two cases is both desirable and a net improvement.